src/components/ClipboardButton/ClipboardButton.tsx

```diff
@@ -81,10 +98,73 @@ const ClipboardButtonComponent = (props: ClipboardButtonComponentProps) => {
 };
 
 export function ClipboardButton(props: ClipboardButtonProps) {
-    const {text, timeout = DEFAULT_TIMEOUT, onCopy, options, ...buttonProps} = props;
+    const {
+        text,
+        timeout = DEFAULT_TIMEOUT,
+        onCopy,
+        options,
+        hasTooltip = true,
+        onMouseEnter,
+        onFocus,
+        ...buttonProps
+    } = props;
+
+    const timerIdRef = React.useRef<number>();
+    const [tooltipCloseDelay, setTooltipCloseDelay] = React.useState<number | undefined>(undefined);
+    const [tooltipDisabled, setTooltipDisabled] = React.useState<boolean>(false);
+
+    React.useEffect(() => window.clearTimeout(timerIdRef.current), []);
+
+    const handleCopy: OnCopyHandler = React.useCallback(
+        (...args) => {
+            onCopy?.(...args);
+            setTooltipDisabled(false);
+            setTooltipCloseDelay(timeout);
+
+            window.clearTimeout(timerIdRef.current);
+
+            timerIdRef.current = window.setTimeout(() => {
+                setTooltipDisabled(true);
+            }, timeout - TOOLTIP_ANIMATION);
+        },
+        [onCopy, timeout],
+    );
+
+    const resetTooltip = React.useCallback(() => {
+        if (tooltipDisabled) {
+            setTooltipDisabled(false);
+            setTooltipCloseDelay(undefined);
+        }
```

Comment:
```suggestion
        if (tooltipDisabled) {
            setTooltipDisabled(false);
        }
        if (tooltipCloseDelay) {
            setTooltipCloseDelay(undefined);
        }
```

Reply:
why?

Reply:
> why?

generally tooltipCloseDelay doesn't depends on tooltipDisabled? Why should we reset it if it's already undefined?

Reply:
it said `if (!showTooltip) {` before, so we only reset if tooltip is made hidden.

Reply:
why? what if we decide not to hide tooltip? it seems we will still need to reset closingDelay.

Reply:
What do you mean by deciding not to hide the tooltip? It has to be hidden even if the user quickly leaves and hovers the button again. We only show the tooltip if it's gone and user hovers the button.